### ISSUE TITLE
I can only add on .gs file per repository?

### ISSUE THREAD

**Issue by GREGG-CHRISTOFFERSON** [leonhartX/gas-github#53]
I tried pushing two different gs files to the same branch on the same repository.  Each file had a different name, but the second file pushed overwrote the 1st file in the master branch, even though they are separate files.

**Comment by leonhartX** [comment]
Thanks for the report.
Can you describe it in detail?
Did you mean you have two GAS project, and you want to sync them to one repo?

**Comment by GREGG-CHRISTOFFERSON** [comment]
Yes that's correct. Also, even within the same GAS project, I may have more
than one .gs file, but any new .gs file I commit to the repository
overwrites the previous.

**Comment by ballatom** [comment]
I encountered the same issue.

I have a code library that contains multiple script files (extension .gs), which I happily pushed to my Bitbucket repository.

I also have a number of spreadsheet files that each have an attached script (that uses the library mentionned earlier).  When I try to push such a script file, it does it but proposes by default to delete the script files of my library.

Ideally, I would like to be able to do the following : 
1. create a directory structure which (if I am not mistaken) is not possible in GAS
2. push all the script files of library 'A' to a directory called 'A'
3. push all the script files of library 'B' to a directory called 'B' 
4. and maybe push all the script files attached to spreadsheets to the root of my repository ?

1 could be done outside of the Extension (using a git client or through the web interface of the repository).

Cheers, Marc

**Comment by leonhartX** [comment]
@ballatom Thanks for the detail.
I've tried the same steps, but can not reproduce the issue.
When you push from GAS to a repo which contains other files, there should be a diff for `DELETE` appeared, did you see?
Below is what I got(I've pushed a `lib/testlib.gs` file from git and then push code from GAS which do not contains the `testlib.gs`)
![2018-01-13 14 28 24](https://user-images.githubusercontent.com/3993518/34903118-0e926892-f86e-11e7-93ef-c55394ab6f19.png)

**Comment by ballatom** [comment]
I get a similar screen when pushing the script of a spreadsheet to a repository that already contains the scripts of a library.
To me, they belong to the same project and hence repository but as far as I see, the extension does not support that out of the box (I would have to push all files to the same repo, at the root and uncheck 'delete' countless times).
If my analysis is correct and others experience the same, I would suggest the following :
- library files are pushed to the chosen repository into a directory named after the library
- spreadsheet files are pushed to the chosen repository into a directory named after the library
Backward compatibility must be taken care of but my experience with git is limited...  It would be something like prompting the user and ask if he wants to use a directory (and remember that choice) named after the file.

What do you think ?  I can help by analysing further and testing.

Marc

**Comment by leonhartX** [comment]
@ballatom Understand your concern.
If your main issue is that you don't want to uncheck those files that under `lib` dir, then you can simply add a ignore pattern from the `config` button like this:

<img width="701" alt="screen shot 2018-01-15 at 10 37 16" src="https://user-images.githubusercontent.com/3993518/34923540-3f98caee-f9e0-11e7-839e-02c05a68a7b1.png">

Then when you push or pull code, the extension will ignore all files that match the regex pattern.
Does this fix your issue? (also @GREGG-CHRISTOFFERSON ?)

**Comment by leonhartX** [comment]
@GREGG-CHRISTOFFERSON  I've reproduced this issue, although I'm not sure whether it is the bug you reported:

1. push file `a.gs`, `b.gs` from a GAS project to github repo `A`
2. push file `c.gs` from another GAS project also to `A`
3. the diff shows you will add `c.gs` and delete `a.gs`,`b.gs`
4. uncheck the deletion of `a.gs` and `b.gs` then push
5. the two files still be deleted.

This is due to a bug only related to push code to github. so the bitbucket issue maybe another one.

**Comment by leonhartX** [comment]
and FYI, if you are opening more than one GAS project in the same browser, the extension may not work as you expected.
see #49

**Comment by ballatom** [comment]
leonhartX

My point is different.

It seems that your extension can only work with the root of the tree.  Hence when you have two libraries or one library and a script file attached to the spreadsheet, it seems you cannot easily push them to the same repo.

Not only will the extension try to delete the 'other' files every time, but they cannot be organised e.g. by storing library A fies in directory A, library B files in directory B and script file at the root.

Does this make sense or do you need more details ?

Kind regards.

Marc

**Comment by leonhartX** [comment]
@ballatom basically understood your issue.
maybe you can workaround with these:
- use a path contains slash to represent a dir(eg: file `lib/some.gs` will synced to file `some.gs` under `lib` dir on github/bitbucket)
- add a ignore pattern in config(eg: `lib/*` will ignore all files under `lib` dir when push/pull)
